pkg/generator/ingress_translator_test.go

```diff
@@ -907,153 +907,18 @@ func TestIngressTranslatorWithUpstreamTLS(t *testing.T) {
 		name: "http2",
 		in: ing("simplens", "simplename", func(ing *v1alpha1.Ingress) {
 			ing.Spec.Rules[0].HTTP.Paths[0].RewriteHost = ""
+			ing.Spec.Rules[0].HTTP.Paths[0].Splits[0].IngressBackend.ServicePort = intstr.FromInt(443)
 		}),
 		state: []runtime.Object{
 			ns("simplens"),
 			svc("servicens", "servicename", func(service *corev1.Service) {
 				service.Spec.Ports = []corev1.ServicePort{{
 					Name:       "http2",
 					TargetPort: intstr.FromInt(8080),
-				}}
-			}),
-			eps("servicens", "servicename"),
-			caSecret,
-		},
-		want: func() *translatedIngress {
-			vHosts := []*route.VirtualHost{
-				envoy.NewVirtualHost(
-					"(simplens/simplename).Rules[0]",
-					[]string{"foo.example.com", "foo.example.com:*"},
-					[]*route.Route{envoy.NewRoute(
-						"(simplens/simplename).Rules[0].Paths[/test]",
-						[]*route.HeaderMatcher{{
-							Name: "testheader",
-							HeaderMatchSpecifier: &route.HeaderMatcher_ExactMatch{
-								ExactMatch: "foo",
-							},
-						}},
-						"/test",
-						[]*route.WeightedCluster_ClusterWeight{
-							envoy.NewWeightedCluster("servicens/servicename", 100, map[string]string{"baz": "gna"}),
-						},
-						0,
-						map[string]string{"foo": "bar"},
-						""),
-					},
-				),
-			}
-
-			return &translatedIngress{
-				name: types.NamespacedName{
-					Namespace: "simplens",
-					Name:      "simplename",
-				},
-				sniMatches: []*envoy.SNIMatch{},
-				clusters: []*v3.Cluster{
-					envoy.NewCluster(
-						"servicens/servicename",
-						5*time.Second,
-						lbEndpoints,
-						true, /* http2 */
-						&envoycorev3.TransportSocket{
-							Name:       wellknown.TransportSocketTls,
-							ConfigType: typedConfig(true /* http2 */),
-						},
-						v3.Cluster_STATIC,
-					),
-				},
-				externalVirtualHosts:    vHosts,
-				externalTLSVirtualHosts: []*route.VirtualHost{},
-				internalVirtualHosts:    vHosts,
-			}
-		}(),
-	}, {
-		name: "http and https",
```

Comment:
These tests are same with `simple` and `http2` test.
Serving always has two ports `http` and `https` so the `svc()` func returns service with the two ports.